app/models/solidus_paypal_braintree/source.rb

```diff
@@ -33,6 +32,29 @@ def imported
       false
     end
 
+    def last_4
+      return self[:last_4] if self[:last_4]
+
+      if braintree_payment_method
+        self[:last_4] = braintree_payment_method.last_4
+        save!
+      end
+
+      self[:last_4]
+    end
+    alias_method :last_digits, :last_4
+
+    def unique_number_identifier
```

Comment:
You need to make sure that we are dealing with a credit cart payment here. Otherwise it will throw this:

```
[Braintree] <paypal-account>
[Braintree]   <billing-agreement-id>B-5FL96061DE115424H</billing-agreement-id>
[Braintree]   <cobranded-card-label nil="true"/>
[Braintree]   <created-at type="datetime">2023-04-24T08:41:25Z</created-at>
[Braintree]   <customer-id>81710890871</customer-id>
[Braintree]   <customer-global-id>Y3VzdG9tZXJfODE3MTA4OTA4NzE</customer-global-id>
[Braintree]   <default type="boolean">true</default>
[Braintree]   <email>team_development-buyer@fcpeuro.com</email>
[Braintree]   <global-id>cGF5bWVudG1ldGhvZF9wcF9uYnhwczBxYg</global-id>
[Braintree]   <image-url>https://assets.braintreegateway.com/payment_method_logo/paypal.png?environment=sandbox</image-url>
[Braintree]   <subscriptions type="array"/>
[Braintree]   <token>nbxps0qb</token>
[Braintree]   <updated-at type="datetime">2023-04-24T08:41:26Z</updated-at>
[Braintree]   <is-channel-initiated type="boolean">false</is-channel-initiated>
[Braintree]   <payer-id>PZQ46JTWXMUEY</payer-id>
[Braintree]   <payer-info nil="true"/>
[Braintree]   <limited-use-order-id nil="true"/>
[Braintree]   <revoked-at nil="true"/>
[Braintree] </paypal-account>

[Module] connection_attempt=1 connection_request_time=0.6426s connection_msg="success"
*** NoMethodError Exception: undefined method `unique_number_identifier' for #<Braintree::PayPalAccount:0x0000000104d10128>

nil
```

Reply:
fixed this myself